src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private double randomValue = Math.abs(random.nextInt());
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Color randomColor = colorSupplier.getRandomColor();
+    private Figure circle = new Circle(randomColor, randomValue);
+    private Figure isoscelesTrapezoid =
+            new IsoscelesTrapezoid(randomColor, randomValue, randomValue, randomValue);
+    private Figure rectangle = new Rectangle(randomColor, randomValue, randomValue);
+    private Figure rightTriangle = new RightTriangle(randomColor, randomValue, randomValue);
+    private Figure square = new Square(randomColor, randomValue);
+    private Figure [] figures =
+            {circle, isoscelesTrapezoid, rectangle, rightTriangle, square, circle};
```

Comment:
You should not create figures and random numbers in this way. You can create figures in private methods, and numbers there too (you can do it separately or together with figures as you like). For in this case, if I call method several times, I will get back figures with the same data, and you have to create new ones each time.